server/channels/app/draft.go

```diff
@@ -107,11 +107,24 @@ func (a *App) getFileInfosForDraft(draft *model.Draft) ([]*model.FileInfo, *mode
 		return nil, nil
 	}
 
-	fileInfos, err := a.Srv().Store().FileInfo().GetByIds(draft.FileIds)
+	allFileInfos, err := a.Srv().Store().FileInfo().GetByIds(draft.FileIds)
 	if err != nil {
 		return nil, model.NewAppError("GetFileInfosForDraft", "app.draft.get_for_draft.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
 	}
 
+	fileInfos := []*model.FileInfo{}
+	for _, fileInfo := range allFileInfos {
+		if fileInfo.PostId == "" && fileInfo.CreatorId == draft.UserId {
+			fileInfos = append(fileInfos, fileInfo)
+		} else {
+			mlog.Debug("Invalid file id in draft", mlog.String("file_id", fileInfo.Id), mlog.String("user_id", draft.UserId))
```

Comment:
Should we just fail hard here? There's no way a bad ID should end up here under normal non-malicious circumstances.

Reply:
I think that'd be a good idea. If we don't expect this sort of thing to come up during normal usage, I don't feel like we should be correcting for that case. Plus, if we error out, it'll be easier to test for.

Reply:
In terms of access policy, I think it would be better to fail here and return a forbidden status.